src/libopensc/pkcs15-cert.c

```diff
@@ -273,26 +273,27 @@ sc_pkcs15_get_extension(struct sc_context *ctx, struct sc_pkcs15_cert *cert,
 			LOG_FUNC_RETURN(ctx, r);
 
 		/* is it the RN we are looking for */
-		if(sc_compare_oid(&oid, type) != 0) {
+		if (sc_compare_oid(&oid, type) != 0) {
 			if (*ext_val == NULL) {
 				*ext_val= val;
 				val = NULL;
 				*ext_val_len = val_len;
+				/* do not free here -- leaking the allocated value */
```

Comment:
This is intention as described in the header of the function:

    if *ext_val is NULL, sc_pkcs15_get_extension will allocate space for ext_val.

which means we are leaking the memory allocated by the call of `sc_asn1_decode()`.